plutus/budget_manager/budget_manager.py

```diff
@@ -0,0 +1,227 @@
+
+                if budget is not None:
+                    with mysql_conn.cursor() as mysql_cursor:
+                        upsert_budget(mysql_cursor, budget, project.project_id, config_type, project.alert_emails)
+
+        else:
+            log.error("Parent folder config verification failed.")
+            metrics.incr("error_count", tags = ["type:misconfig", f"parent_id:{parent_id}", f"config_type:{config_type}"])
+            sys.exit(1)
+
+
+    # TODO - Have a google support ticket, currently specifying more than one label returns a union not an intersection
+    for label_dict in budget_dict['labels']:
+        config_type = PLUTUS_CONFIG_TYPE_LABEL
+
+        if verify_labels_yaml(label_dict):
+
+            labels_filter = {}
+            for row in label_dict['label_list']:
+                for key in row:
+                    labels_filter[f"labels.{key}"] = row[key]
+
+            # TODO - remove this count once google support ticket resolved
+            count = 0
+            for p in resource_manager_client.list_projects(filter_params=labels_filter):
+                # TODO - rm me
+                count += 1
+
+                # Every iteration, we overwrite the 'project_id' key for each project that matches labels
+                label_dict['project_id'] = p.project_id
+                project = ProjectBudget(label_dict, config_type, billing_account_id, default_pubsub_topic)
+
+                # TODO - add back more labels to config and test once google ticket resolved
+                budget = gcp.get_and_update_or_create_budget(project)
+
+                if budget is not None:
+                    with mysql_conn.cursor() as mysql_cursor:
+                        upsert_budget(mysql_cursor, budget, project.project_id, config_type, project.alert_emails)
+
+            # TODO - remove this after ticket resolved
+            log.info(count)
+
+        else:
+            log.error("Labels config verification failed.")
+            metrics.incr("error_count", tags = ["type:misconfig", f"config_type:{config_type}"])
+            sys.exit(1)
+
+    # =-=-=-=-=-=-=-=-=-=-=-=-==-=- DEFAULTS START HERE =-=-=-=-=-=-=-=-=-=- #
+
+    default_dict = budget_dict['default']
+    config_type = PLUTUS_CONFIG_TYPE_DEFAULT
+
+    if verify_default_yaml(default_dict):
+        # Query for all projects. If no budget exists for project, add a default budget
+        for p in resource_manager_client.list_projects():
+            project_id = p.project_id
+            project_number = gcp.get_project_number(project_id)
+            if project_number is not None:
+                budgets = gcp.get_budgets_by_project(project, project_number)
+                if len(budgets) == 0:
+                    # No budget exists for this project, so we create one
+                    default_dict['project_id'] = project_id
+                    log.info(f"will create default budget for plutus-default-{project_id}")
```

Comment:
The wording around this log is confusing, maybe something like "Default budget created for project..." or if this is to capture start of the action then maybe "Attempting to create default budget for ..."

Reply:
rm'ing default logic for now